### DISCUSSION TITLE
Role Based Authentication

### DISCUSSION THREAD

**Discussion by diogoribeirodev** [nextauthjs/next-auth#6467] [Help]
### Question 💬

Hi, i see that NextAuth supports role based login,
https://next-auth.js.org/tutorials/role-based-login-strategy

I can see on the docs how to access the role and return it on session, but i Don't see how i can pass the role i want on signup, is it possible when passing the signup() to also give the role i want it to add to the user? Or do i need to create my own callback function that when a user signups it updates the user on the database ( via prisma ), if so how should i pass the role i want? ( If using the second option)

### How to reproduce ☕️

model User {
  id            String    @id @default(cuid())
  name          String?
  email         String?   @unique
  emailVerified DateTime?
  image         String?
  role          String?  // New Column
  accounts      Account[]
  sessions      Session[]
}

callbacks: {
  async session({ session, token, user }) {
    session.user.role = user.role; // Add role value to user object so it is passed along with session
    return session;
  }
},

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by NorkzYT** [comment]
Hello mate,

Change the `role` column in the `prisma schema`  to `role String? @default("user")` to make all new users have a default role of `user`. Replace the `user` value with the desired default role.

For example.

```prisma
model User {
id String @id @default(cuid())
name String?
email String? https://github.com/unique?type=source
emailVerified DateTime?
image String?
role String? @default("user")
accounts Account[]
sessions Session[]
}
```

**Reply by diogoribeirodev** [reply]
i know i can do that, what i want is to pass a custom role to the user when he signups,

**Reply by NorkzYT** [reply]
Ah, understood. Unfortunately, I do not know how to do that. I did find the following issue.

https://github.com/nextauthjs/next-auth/discussions/805#discussioncomment-351960

**Comment by MarvingMoreton** [comment]
Hello @Diogovski, have you found any solution?

I am basically looking for the same thing!!

Thanks
